### PR TITLE
New posts

### PR DESCRIPTION
**Pull request by mahim23**
Blog posts for 30th Oct

### CODE REVIEW

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+## Containers
+
+An image is a lightweight, stand-alone, executable package that includes everything needed to run a piece of software, including the code, a runtime, libraries, environment variables, and config files.
+
+A container is a runtime instance of an image—what the image becomes in memory when actually executed.
+It runs completely isolated from the host environment by default, only accessing host files and ports if
+configured to do so.
+
+Containers run apps natively on the host machine’s kernel. They have better performance characteristics
+than virtual machines that only get virtual access to host resources through a hypervisor. Containers can
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+In a way, Docker is a bit like a virtual machine. But unlike a virtual machine, rather than creating a whole virtual operating system, Docker allows applications to use the same Linux kernel as the system that they're running on and only requires applications be shipped with things not already running on the host computer. This gives a significant performance boost and reduces the size of the application.
+
+## Containers
+
+An image is a lightweight, stand-alone, executable package that includes everything needed to run a piece of software, including the code, a runtime, libraries, environment variables, and config files.
+
+A container is a runtime instance of an image—what the image becomes in memory when actually executed.
+It runs completely isolated from the host environment by default, only accessing host files and ports if
+configured to do so.
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+Containers run apps natively on the host machine’s kernel. They have better performance characteristics
+than virtual machines that only get virtual access to host resources through a hypervisor. Containers can
+get native access, each one running in a discrete process, taking no more memory than any other executable.
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+## Containers vs. Virtual Machines
+
+![Containers vs VM](/blog/assets/img/Docker/docker-vm-container.png)
+
+**Virtual Machines** run guest operating systems—note the OS layer in each box. This is resource intensive,
+and the resulting disk image and application state is an entanglement of OS settings, system-installed
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+dependencies, OS security patches, and other easy-to-lose, hard-to-replicate ephemera.
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+**Containers** can share a single kernel, and the only information that needs to be in a container image is
+the executable and its package dependencies, which never need to be installed on the host system. These
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+processes run like native processes, and you can manage them individually by running commands like docker
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+ps—just like you would run ps on Linux to see active processes. Finally, because they contain all their
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+dependencies, there is no configuration entanglement; a containerized app **“runs anywhere”**.
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+## Who is Docker for?
+
+Docker is a tool that is designed to benefit both developers and system administrators, making it a part
+of many DevOps (developers + operations) toolchains. For *developers*, it means that they can focus on writing
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+code without worrying about the system that it will ultimately be running on. It also allows them to get a
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+head start by using one of thousands of programs already designed to run in a Docker container as a part of
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+their application. For *operations*, Docker gives flexibility and potentially reduces the number of
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+systems needed because of its small footprint and lower overhead.
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+
+## In a nutshell, here's what Docker can do for you: 
+
+It can get more applications running on the same hardware than other technologies; it makes it easy for
+developers to quickly create ready-to-run containered applications; and it makes managing and deploying
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+applications much easier. Put it all together and I can see why Docker is riding the hype cycle as fast
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+as I can recall ever seeing an enterprise technology go.
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-Intoduction-to-Docker.md [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+
+## Setup 
+
+Docker is available on multiple platforms, on cloud and on-premises. It can be downloaded and installed by
+following the instructions [here](https://docs.docker.com/engine/installation/)
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-Intoduction-to-Docker.md]: Why is this going to the next line. It's one full sentence right?

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+categories:
+- Piston
+github_username: 'meghbhalerao'
+---
+
+The LIGO (Laser Interferometer Gravitational-Wave Observatory) Experiment – A completely different perspective of studying the universe.
+
+**Einstein’s Prediction**.
+
+<img src="/blog/assets/img/the-ligo-experiment/image1.jpeg" width="270" height="294" />Over a century ago, Albert Einstein proposed the idea of something called a Gravitational Wave which was a part of his General Theory of Relativity. The concept of Gravitational waves suggests that, the force of gravity is not an actual force but rather, a distortion in the fabric of spacetime. This was just an idea, as there was no experimental evidence to support Einstein’s claim.
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Leave some space between image and text!

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+tags:
+- LIGO
+categories:
+- Piston
+github_username: 'meghbhalerao'
+---
+
+The LIGO (Laser Interferometer Gravitational-Wave Observatory) Experiment – A completely different perspective of studying the universe.
+
+**Einstein’s Prediction**.
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Make this as a h2 tag `##`

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+
+<img src="/blog/assets/img/the-ligo-experiment/image1.jpeg" width="270" height="294" />Over a century ago, Albert Einstein proposed the idea of something called a Gravitational Wave which was a part of his General Theory of Relativity. The concept of Gravitational waves suggests that, the force of gravity is not an actual force but rather, a distortion in the fabric of spacetime. This was just an idea, as there was no experimental evidence to support Einstein’s claim.
+
+**What is a gravitational wave?**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Make this as a h2 tag `##`

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+
+A gravitational wave is much like an ordinary ripple caused by any distortion in a medium. The only difference here is that the ripple is in the medium(fabric) of spacetime. A gravitational wave originates from an accelerating mass in spacetime. Some phenomena that could trigger a gravitational wave are two black holes rotating in a dead spiral, collision of neutron stars, or even a ball dropping off your hand. The gravitational waves compress space in one direction and expand it in the orthogonal direction.
+
+**Conceiving the idea for an experimental setup for detecting gravitational waves.**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Make this as a h2 tag `##`

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 23]
```diff
@@ -0,0 +1,56 @@
...
+
+During the early 1970s an ambitious professor at the MIT named Rainer Weiss first proposed the idea of an experimental setup to detect gravitational waves. His idea was funded by the US national science foundation, and the very first version of the LIGO was setup in 1983. This very primitive version of the LIGO which had arms of length 1km failed to produce any concrete evidence of gravitational waves. It was not until 2014 that the advanced LIGO was made fully operational. Rainer Weiss recently won the Nobel prize for Physics in 2017.
+
+<img src="/blog/assets/img/the-ligo-experiment/image2.png" width="388" height="259" />
+
+**When were the first gravitational waves detected?**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Make this as a h2 tag `##`

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+
+On the 14<sup>th</sup> of September, Rainer Weiss noticed something peculiar on the usual activity log of the LIGO. He instantly contacted the field engineers who were present at the observatory and Weiss was completely dumbfounded. Gravitational waves from the merger of two black holes, each of nearly 30 solar masses, 1.5 billion light years away were detected for a fraction of a second.
+
+**So, how does the LIGO exactly detect these “Gravitational waves”?**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Make this as a h2 tag `##`

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+
+1. The two light beams being 180 degrees out of phase will cancel out each other and nothing will be detected by the sensor – this will happen when the two light beams traverse the exact same path.
+
+2. The two light beams have a phase difference other than 180 degrees. In this case the two beams will not exactly cancel out each other hence leaving a difference in the amplitudes. This small difference in the amplitudes is detected by the sensor, giving information about the gravitational wave that struck the LIGO apparatus.
+<img src="/blog/assets/img/the-ligo-experiment/image3.jpeg" width="441" height="296" />
+
+**The principle doesn’t look that complicated, then why did it take so many years to get it operational?**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Make this as a h2 tag `##`

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+
+<img src="/blog/assets/img/the-ligo-experiment/image4.png" width="392" height="238" />
+
+**What role did India play in the LIGO experiment?**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Make this as a h2 tag `##`

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 27]
```diff
@@ -0,0 +1,56 @@
...
+
+The IUCAA (Inter University Centre for Astronomy and Astrophysics) at Pune played an important role in the research and development of the experiment. About 40 scientists from India participated in the experiment. India has taken a further step ahead and has decided on building a LIGO apparatus at a project cost of 1000 Cr.
+
+**How does the gravitational wave picture change our understanding of the cosmos?**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Make this as a h2 tag `##`

File: blog_src/_posts/2017-10-30-wireless-power-transmission.md [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+layout: post
+title: "Wireless Power Transmission"
+author_github: vansjyo
+date: 2017-10-30 10:10:08
+image: '/assets/img/'
+description: 'Boon to the next generation, transmitting power directly without radioactive methods.'
+tags:
+- Power
+categories:
+- CAS
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-wireless-power-transmission.md]: This is Piston na?

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 29]
```diff
@@ -0,0 +1,56 @@
...
+layout: post
+title: "The LIGO Experiment"
+author_github: meghbhalerao
+date: 2017-10-30 10:10:09
+image: '/assets/img/'
+description: 'Every flight begins with a fall'
+tags:
+- LIGO
+categories:
+- Piston
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: This is Diode na?

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 30]
```diff
@@ -0,0 +1,56 @@
...
+
+The most important aspect of this observatory is the sensitivity of the measuring instruments that are used in the setup. The gravitational waves come from hundreds of millions of light years away and hence their intensity is very weak. The LIGO measures lengths of the order 10<sup>-21</sup> of a meter. This length is roughly the size of one thousandths of the diameter of a typical atomic nucleus. To get such sensitive measurements the LIGO uses the purest of laser light and mirrors of the highest quality. The mirrors weigh about ninety kilos each and are highly polished. The reason that the LIGO is placed in such isolated locations is to prevent any stray noise from interfering with the data. The LIGO also uses technology such as internal seismic isolation to prevent any stray noise from corrupting the results obtained. The two LIGO setups are placed so far off from each other so that the common portion of the response of the sensors can be filtered out thus eliminating the chance of any common noise. This makes the LIGO truly an engineering marvel.
+
+<img src="/blog/assets/img/the-ligo-experiment/image4.png" width="392" height="238" />
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Use markdown image syntax instead!

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+
+1. The two light beams being 180 degrees out of phase will cancel out each other and nothing will be detected by the sensor – this will happen when the two light beams traverse the exact same path.
+
+2. The two light beams have a phase difference other than 180 degrees. In this case the two beams will not exactly cancel out each other hence leaving a difference in the amplitudes. This small difference in the amplitudes is detected by the sensor, giving information about the gravitational wave that struck the LIGO apparatus.
+<img src="/blog/assets/img/the-ligo-experiment/image3.jpeg" width="441" height="296" />
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Use Markdown image syntax instead

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md [hunk 32]
```diff
@@ -0,0 +1,56 @@
...
+
+**Conceiving the idea for an experimental setup for detecting gravitational waves.**
+
+During the early 1970s an ambitious professor at the MIT named Rainer Weiss first proposed the idea of an experimental setup to detect gravitational waves. His idea was funded by the US national science foundation, and the very first version of the LIGO was setup in 1983. This very primitive version of the LIGO which had arms of length 1km failed to produce any concrete evidence of gravitational waves. It was not until 2014 that the advanced LIGO was made fully operational. Rainer Weiss recently won the Nobel prize for Physics in 2017.
+
+<img src="/blog/assets/img/the-ligo-experiment/image2.png" width="388" height="259" />
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Use Markdown image syntax instead

File: blog_src/_posts/2017-10-30-the-ligo-experiment.md (hunk 19 above)
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-the-ligo-experiment.md]: Use Makrdown image syntax instead!

File: blog_src/_posts/2017-10-30-wireless-power-transmission.md [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+
+Scientists have tried to develop methods of wireless power transmission that could cut the clutter or lead to clean sources of electricity. 
+
+The attempt to transmit power wirelessly is not new. A revolutionary attempt towards this was made by Nicola Tesla by inventing the wonder of physics, The Tesla Coil, with an aim to provide electricity without wires but it failed miserably due to many drawbacks which made it highly uneconomical.
+
+*Quick Physics:* The Tesla coil creates an electric field that pushes electrons through the light bulb. This is the same way the lights in your house work, except in your house, the electricty comes through a wire instead of through the air.
+
+A Tesla coil makes current electricity; the charges are flowing.  If a fluorescent light bulb is held near the coil, the electricity will then go through the light bulb to get to the ground, which makes it light up. 
+
+### The modern Techniques developed so far
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-wireless-power-transmission.md]: Make this as a h2 tag `##`

File: blog_src/_posts/2017-10-30-wireless-power-transmission.md [hunk 34]
```diff
@@ -0,0 +1,43 @@
...
+
+**Short distance induction :**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-wireless-power-transmission.md]: Make this as a h3 tag `###`

File: blog_src/_posts/2017-10-30-wireless-power-transmission.md [hunk 35]
```diff
@@ -0,0 +1,43 @@
...
+These methods can reach at most a few centimetres The action of an electrical transformer is the simplest instance of wireless energy transfer. The primary and secondary circuits of a transformer are electrically isolated from each other. The transfer of energy takes place by electromagnetic coupling through a process known as mutual induction. This method is known as inductive coupling.
+
+**Moderate distance :**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-wireless-power-transmission.md]: Make this as a h3 tag `###`

File: blog_src/_posts/2017-10-30-wireless-power-transmission.md [hunk 36]
```diff
@@ -0,0 +1,43 @@
...
+Since a magnetic field spreads in all directions, making a larger one would waste a lot of energy. An efficient way to transfer power between coils separated by a few meters is that we could extend the distance between the coils by adding resonance to the equation, ie. the frequency at which an object  naturally vibrates. It's easy to get objects to vibrate at their resonant frequency. This process is known as "Resonant inductive coupling" 
+
+**Long-distance Wireless Power:**
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-wireless-power-transmission.md]: Make this as a h3 tag `###`

File: blog_src/_posts/2017-10-30-wireless-power-transmission.md [hunk 37]
```diff
@@ -0,0 +1,43 @@
...
+The secret to the SHARP's(the unmanned plane)  long flight time was a large, ground-based microwave transmitter. The SHARP's circular flight path kept it in range of this transmitter. A large, disc-shaped rectifying antenna, or rectenna, just behind the plane's wings changed the microwave energy from the transmitter into direct-current (DC) electricity. Because of the microwaves' interaction with the rectenna, the SHARP had a constant power supply as long as it was in range of a functioning microwave array. 
+
+### Conclusion
```
Comment (salman-bhai) [inline blog_src/_posts/2017-10-30-wireless-power-transmission.md]: Make this as a h2 tag `##`

**Comment by mahim23** [comment]
What should I change the pull request's description and title to?

**Pull request merged**